### DISCUSSION TITLE
Flair API for querying temp sensors from Ecobee is way too slow

### DISCUSSION THREAD

**Discussion by seanmahrt** [RobertD502/home-assistant-flair#71] [General]
I asked flair why their sensors are so slow responding to heating events (when in enhanced circulation mode).  they said they have roughly a 5 minute polling time to Ecobee, which isn't going to cut it...

So, I'm starting to think about just wrapping my own control around the flair vents and use Robert's great library to set vent position (open/closed).  it's a giant PITA to build basically generic thermostats around each room's set of vents, but I'm not sure if there's a cleaner way. Threshold helper is out, since it doesn't take variables in upper/lower.  maybe a state machine.  But I have 5 rooms.  trying to keep it to a managable pattern...

Also, apparently Ecobee doesn't handle pre-heat/pre-cool well to the flair vents.  and it doesn't seem to expose that it's in that mode.  so for the entire pre-heat cycle before my trigger time all the vents are closed..  ugh.

so much for "easy integration".

Making sure I'm not reinventing the wheel here.

Sean

**Comment by RobertD502** [comment]
Yeah, polling Ecobee isn't the best.  If you don't want Flair to automatically control the vents, you can place the vents in "Flair rooms" (within the Flair app) that don't have a compatible sensor or flair puck associated with them, which will allow for them to be controlled manually by you. This would require you to build out the logic yourself though and control at what temperature vents open/close. Since you have an Ecobee, you can integrate it into Home Assistant directly via HomeKit, and then use the ecobee climate entity to know the exact moment a heating event is occurring/use this information in your automation.

Since it requires more complex logic, I use NodeRed in such scenarios to utilize temp sensors that are not supported directly by Flair while replicating their +- 1 degree hysteresis logic.

Example: I control a vent based on the temp I get from a zigbee sensor in the room. I set an upper and lower temp (setpoint + 1 for upper, setpoint - 1 for lower) to replicate the hysteresis seen in auto mode. Then it diverges based on if the current temp reading is <= the lower temp or >= the upper temp. I then check to see if my HVAC system is currently in heating or cooling mode:

**If current temp <= lower temp:**
1. If in `cooling` mode, close the vent if it is open
2. If in `heating` mode, open the vent if it is closed

**If current temp >= lower temp:**
1. If in `cooling` mode, open the vent if it is closed
2. If in `heating` mode, close the vent if it is open

![image](https://github.com/RobertD502/home-assistant-flair/assets/52541649/e3f9ec05-9b6b-4519-9d70-621a1cc0d9e0)

I don't use the enhanced circulation mode myself, so that isn't included in what I am doing. You'll have to replicate the enhanced circulation logic into your automation.
